### PR TITLE
One canvas

### PR DESCRIPTION
**Pull request by ashleychangg**

### CODE REVIEW

File: common/public/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+data/
```
Comment (mhendri) [inline common/public/.gitignore:1]: Why is there a .gitignore file in common/public?

File: common/public/js/pseudo-image.js [hunk 2]
```diff
@@ -83,7 +82,7 @@ function initializeScales() {
 		.attr("width", constants.width + constants.margin.left + constants.margin.right + 50)
 		.attr("height", constants.height + constants.margin.top + constants.margin.bottom)
 		.style("float","left")
-		.append("g")
+		.append("g").attr('cursor', 'pointer')
```
Comment (mhendri) [inline common/public/js/pseudo-image.js:85]: Generally, the cursor pointer represents that the item being hovered on is a link. I don't know that it is appropriate or necessary for indicating a zoom feature.

File: common/public/js/pseudo-image.js [hunk 3]
```diff
@@ -173,11 +171,10 @@ function getControlPanelValues() {
 	];
 }
 
-
 // 3) CANVAS INTERACTING FUNCTIONS
 // 3a) Binds data to canvas
-function databind(drawData, product = null, r = "670 nm", g = "554 nm", b = "469 nm", brightness = 1.0) {
-	// let t0 = performance.now(); // uncomment to test performance
+function databind(allFileData, product = null, r = "670 nm", g = "554 nm", b = "469 nm", brightness = 1.0) {
```
Comment (mhendri) [inline common/public/js/pseudo-image.js:176]: Please confirm the performance times using the performance.now() functionality.

File: common/public/js/pseudo-image.js [hunk 4]
```diff
@@ -613,6 +615,7 @@ function plotScans() {
 		data = [],
 		bindData = [],
 		currentScans = {};
+		scansList = [];
```
Comment (mhendri) [inline common/public/js/pseudo-image.js:618]: Delete?

**Comment by mhendri** [comment]
Also, please remove the DS_Store file from the repo

File: common/public/js/pseudo-image.js (hunk 2 above)
Comment (ashleychangg) [reply common/public/js/pseudo-image.js:85]: Ok I got rid of it

File: common/public/js/pseudo-image.js [hunk 5]
```diff
@@ -195,8 +195,7 @@ function databind(allFileData, product = null, r = "670 nm", g = "554 nm", b = "
 
 	// for every element in drawData (195 elements)
 	drawData = allFileData[0];
-	for (var outerIndex = 0, index = 0;outerIndex < allFileData.length && index < drawData.length; index++, outerIndex=(index==drawData.length)?outerIndex+1:outerIndex, index=(index==drawData.length)?index=0:index){
-		//j++, i=(j==10)?i+1:i,j=(j==10)?j=0:j) 		
+	for (var outerIndex = 0, index = 0;outerIndex < allFileData.length && index < drawData.length; index++, outerIndex=(index==drawData.length)?outerIndex+1:outerIndex, index=(index==drawData.length)?index=0:index){
```
Comment (mhendri) [inline common/public/js/pseudo-image.js:198]: Could you just add a clear and descriptive comment here about what this for loop is doing, and how it works. It's the kind of thing you come back to in a few months and its like "what the hell does this do?!"

File: common/public/js/pseudo-image.js [hunk 6]
```diff
@@ -193,21 +193,21 @@ function databind(allFileData, product = null, r = "670 nm", g = "554 nm", b = "
 
 	globals.custom = d3.select(customBase); // this is our svg replacement
 
-	// for every element in drawData (195 elements)
-	//efficiently loops through all files to be drawn (reduces code complexity by replacing nested loop)
-	drawData = allFileData[0]; //initialize drawData before loop
-	for (var outerIndex = 0, index = 0;outerIndex < allFileData.length && index < drawData.length; index++, outerIndex=(index==drawData.length)?outerIndex+1:outerIndex, index=(index==drawData.length)?index=0:index){		
-		if(index == 0){ //only once after each interation of the outer loop
-			drawData = allFileData[outerIndex];
+	//for each of the 195 elements in each of the 70 elements in each file (typically 20 files)
```
Comment (mhendri) [inline common/public/js/pseudo-image.js:196]: This is still a clunky way to describe this and requires some mental gymnastics for me to make sense of it. Use plain language and start from the other direction: "For each file, of which there are typically 20, get the data for the product to be graphed. The product array has an inner array that represents one scan. Each scan has an array of 195 elements representing each sector of the scan. Each of these 195 elements has an array of length 9, representing the 9 bands of the RSP." *Something LIKE this*

File: common/public/js/pseudo-image.js [hunk 7]
```diff
@@ -193,21 +193,21 @@ function databind(allFileData, product = null, r = "670 nm", g = "554 nm", b = "
...
+	//efficiently loops through all data to be drawn 
+	drawData = allFileData[0]; 
+	for (var file = 0, element = 0;file < allFileData.length && element < drawData.length; element++, file=(element==drawData.length)?file+1:file, element=(element==drawData.length)?element=0:element){		
+		if(element == 0){ 
+			drawData = allFileData[file]; //each file has length of 70
 		}
-		globals.custom.selectAll(`custom.rect${index}`)
-				.data(drawData[index])
+		globals.custom.selectAll(`custom.rect${element}`)
+				.data(drawData[element]) //195 elements
```
Comment (mhendri) [inline common/public/js/pseudo-image.js:204]: who has 195 elements, drawData, or element?

File: common/public/js/pseudo-image.js [hunk 8]
```diff
@@ -193,21 +193,21 @@ function databind(allFileData, product = null, r = "670 nm", g = "554 nm", b = "
...
 				.enter()
 				.append('custom')
 				.attr('class', 'rect')
 				.attr('width', 1)//constants.width / globals.scan_count)
 				.attr('height', 1)//constants.height / constants.sector_count)
-				.attr('x', (d) => index + (outerIndex * 70))//s * constants.width / globals.scan_count)
+				.attr('x', (d) => element + (file * 70))//s * constants.width / globals.scan_count)
```
Comment (mhendri) [inline common/public/js/pseudo-image.js:210]: 70 shouldn't be hard coded right?

**Pull request merged**